### PR TITLE
Add Okta to auth providers

### PR DESCRIPTION
**Pull request by NoahC5**
The purpose of this pull request is to add in Okta as an authentication provider. This PR is WIP, I made a draft so I could get feedback and some questions answered. This is also my first pull request made to an open-source project, so I appreciate any general PR advice/criticism!

### Questions/Incomplete parts

**Okta Auth Client**

- From what I can see, the `signup` method cannot be mapped to the [Okta Auth JavaScript SDK](https://github.com/okta/okta-auth-js#okta-auth-javascript-sdk) because the accounts are created by organization admins in Okta's dashboard. I'm not sure what to do for this scenario.

**Okta Decoder**

- I'm not sure if I'm passing the decoder the right token. Currently I am passing the decoder the `accessToken` but the `idToken` reveals more information about the current user. Right now the `context.currentUser` on the api side is just given the access token claims which doesn't seem right to me but I am able to get the email address of the current user from this information.
- I'm not sure how to handle the event of a token expiring. 

**Okta CLI setup command**

- An error occurs when I try to run the `yarn rw setup auth okta` on a test project. This error occurs during the `Adding required web packages...` step and outputs `Command failed with exit code 1: yarn workspace web add @okta/okta-auth-js @redwoodjs/auth@0.49.1` I understand that this may be caused by the way I added the dependencies to the framework but I am unsure about how to add the properly in a way that fixes this issue if it is even the cause.

### To test locally

Follow the instructions given on the [Okta Auth JavaScript SDK Documentation](https://github.com/okta/okta-auth-js#getting-started) under the 'Getting Started' section. 

In your .env you'll need to have these variables filled in:

- OKTA_ISSUER
- OKTA_CLIENT_ID
- OKTA_REDIRECT_URI
- OKTA_AUDIENCE
- OKTA_DOMAIN

The default auth.js setup is used but won't be automatically generated because the setup command doesn't work.

### CODE REVIEW

**Comment by jtoar** [comment]
Thanks for your patience @NoahC5! I tagged @dthyresson on this, he'll review this and help you get this one in.

**Review by dthyresson — APPROVED** [review]
LGTM let's gt in latest and move out of draft and in a canary release, also update Playground Auth to test and verify.

File: packages/cli/src/commands/setup/auth/providers/okta.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// the lines that need to be added to App.{js,tsx}
+export const config = {
+  imports: [`import { OktaAuth } from '@okta/okta-auth-js'`],
+  init: `
+  const okta = new OktaAuth({
+    issuer: 'process.env.OKTA_ISSUER',
```
[the remark this answers is not in the document]
Comment (NoahC5) [reply packages/cli/src/commands/setup/auth/providers/okta.js]: This should be fixed in the most recent version.

**Comment by jtoar** [comment]
Great work @NoahC5! Mind resolving the merge conflicts so we can update from main and adding docs so that everyone knows about this new provider?

**Comment by NoahC5** [comment]
The merge conflicts should be resolved now. Should the documentation be updated in this pull request or should I create a new pull request for the documentation.

**Comment by jtoar** [comment]
@NoahC5 you can update the docs here in this PR. The one that you probably want to update is here: https://github.com/redwoodjs/redwood/blob/main/docs/docs/authentication.md.

**Comment by NoahC5** [comment]
@jtoar Sounds good!

**Comment by NoahC5** [comment]
What is the timeline for getting this merged? Is there more that has to be done?

**Comment by jtoar** [comment]
Hey @NoahC5, I just rebased again and committed a few fixes. Tried to make the push to get it merged today but looking like it'll be tomorrow. The last for me to do now is to manually test it. Have you done that yourself already?

**Comment by jtoar** [comment]
Gitpod isn't working too well, and given that we just merged but haven't released https://github.com/redwoodjs/redwood/pull/4880 yet, I'm bending over backwards to test this locally. So I'm going to merge this and test when it's in canary.

Sorry this one took so long to land! Thanks for everything @NoahC5—your patience and persistence especially. 🙏

**Review by jtoar — APPROVED** [review]

**Pull request merged**